docs/src/getting_started.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Getting Started Guide
+slug: /getting_started
+---
+
+# Getting Started with Charmil
+
+This guide assumes that you already have a working Go environment, if not then please refer to
+[this page](https://golang.org/doc/install) first.
+
+## Charmil SDK
+
+This is the component responsible for constructing modular, multi-repo Golang CLIs with Cobra while providing a variety of high-level solutions for typical challenges like configuration, internationalization, etc.
+&nbsp;
+
+- ### Installation:
+
+  First use `go get` to install the latest tagged release of the framework.
+  The following command will install the framework along with its dependencies:
+
+  ```bash
+  $ go get -u github.com/aerogear/charmil
+  ```
+
+  Next, import the required package into your project:
+
+  - For the config management package:
+
+    ```go
+    import "github.com/aerogear/charmil/core/config"
+    ```
+
+  - For utils packages (eg. color, localize, logging, etc.):
+
+    ```go
+    import "github.com/aerogear/charmil/core/utils/{name_of_the_required_package}"
+    ```
+
+- ### Usage:
+
+  For usage instructions on Charmil SDK, please refer to the docs [over here](./).
+
+## Charmil Validator
+
+- ### Installation:
+
+  First use `go get` to install the latest tagged release of the framework.
+  The following command will install the framework along with its dependencies:
+
+  ```bash
+  $ go get -u github.com/aerogear/charmil
+  ```
+
+  Next, import the Charmil Validator package into your project:
+
+  ```go
+  import "github.com/aerogear/charmil/validator"
+  ```
+
+- ### Usage:
+
+  For usage instructions on Charmil Validator, please refer to [this link](./charmil_validator.md#how-to-use).
+
+## Charmil Starter
```

Comment:
I think this should be first. There is no point for anyone reading core or validator as starter should have all integrated and ready.